app/src/main/java/com/internship/move/feature/map/MapFragment.kt

```diff
@@ -1,47 +1,261 @@
+import com.google.android.gms.maps.model.BitmapDescriptorFactory
+import com.google.android.gms.maps.model.LatLng
+import com.google.android.gms.maps.model.MapStyleOptions
+import com.google.android.gms.maps.model.MarkerOptions
 import com.internship.move.R
 import com.internship.move.databinding.FragmentMapBinding
 import com.internship.move.feature.authentication.AuthenticationViewModel
-import com.internship.move.feature.onboarding.OnboardingViewModel
 import com.internship.move.utils.logTag
 import com.zhuinden.fragmentviewbindingdelegatekt.viewBinding
 import org.koin.androidx.viewmodel.ext.android.viewModel
+import java.util.*
 
-class MapFragment : Fragment(R.layout.fragment_map) {
-
+class MapFragment : Fragment(R.layout.fragment_map), OnMapReadyCallback {
     private val binding by viewBinding(FragmentMapBinding::bind)
-    private val onboardingViewModel: OnboardingViewModel by viewModel()
     private val authViewModel: AuthenticationViewModel by viewModel()
+    private val mapViewModel: MapViewModel by viewModel()
+    private lateinit var scooterMap: GoogleMap
+    private lateinit var fusedLocationClient: FusedLocationProviderClient
+    private val permissionId = NUMBER_OF_PERMISSIONS
+    private var doubleBackPressed = false
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
         authViewModel.getCurrentUser()
         logTag("MAP_USER", authViewModel.getCurrentUser().toString())
-        initListeners()
+
+        binding.scooterMapView.onCreate(savedInstanceState)
+        binding.scooterMapView.onResume()
+        fusedLocationClient = LocationServices.getFusedLocationProviderClient(requireContext())
+        getLocation()
         initViews()
+        initButtons()
+        initObservers()
+    }
+
+    private fun initViews() {
+        binding.scooterMapView.getMapAsync(this)
+    }
+
+    private fun initObservers() {
+        mapViewModel.scooterList.observe(viewLifecycleOwner) { scooterList ->
+            if (scooterList.isNotEmpty())
+                scooterList.forEach { scooterItem ->
+                    scooterMap.addMarker(
+                        MarkerOptions().position(
+                            LatLng(
+                                scooterItem.location.coordinates[1],
+                                scooterItem.location.coordinates[0]
+                            )
+                        ).title(scooterItem._id)
+                            .icon(bitmapDescriptorFromVector(R.drawable.ic_map_pin))
+                    )
+                }
+        }
+
+        mapViewModel.currentLocation.observe(viewLifecycleOwner) { location ->
+            scooterMap.moveCamera(CameraUpdateFactory.newLatLng(location))
+            mapViewModel.getScooters(location.latitude.toFloat(), location.longitude.toFloat())
+        }
+    }
+
+    private fun bitmapDescriptorFromVector(vectorResId: Int): BitmapDescriptor {
```

Comment:
I would bet that you copy pasted this method from Java and Android Studio transformed it to kotlin. Am I right? 😁 
This isn't a problem, we do it all the time. But, Android Studio kotlin to java transformation isn't perfect and many times leaves us with lint warnings that should be fixed:
`vectorDrawable.getIntrinsicWidth()` should be replaced with `vectorDrawable.intrinsicWidth` (similar for height)
Replace `!!` calls with `?.` or null checks

Reply:
Use `ResourcesCompat.getDrawable(..)` instead of `ContextCompat` to avoid crashes